mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,270 @@
+          if (!predictor.isPresent()) {
+            return yieldMissingPredictorError(response);
+          }
+
+          try {
+            String result = evaluateRequest(predictor.get(), request);
+            response.status(HTTP_RESPONSE_CODE_SUCCESS);
+            return result;
+          } catch (PredictorEvaluationException e) {
+            response.status(HTTP_RESPONSE_CODE_SERVER_ERROR);
+            String errorMessage = e.getMessage();
+            return getErrorResponseJson(errorMessage);
+          } catch (Exception e) {
+            e.printStackTrace();
+            response.status(HTTP_RESPONSE_CODE_SERVER_ERROR);
+            String errorMessage = "An unknown error occurred while evaluating the model!";
+            return getErrorResponseJson(errorMessage);
+          }
+        });
+
+    this.activeService = Optional.of(newService);
+  }
+
+  /**
+   * Stops the scoring server
+   *
+   * @throws IllegalStateException If the server is not active
+   */
+  public void stop() {
+    if (activeService.isPresent()) {
+      activeService.get().stop();
+      activeService = Optional.empty();
+    } else {
+      throw new IllegalStateException("Attempted to stop the server that is not active!");
+    }
+  }
+
+  /** @return `true` if the server is active (running), `false` otherwise */
+  public boolean isActive() {
+    return activeService.isPresent();
+  }
+
+  private String yieldMissingPredictorError(Response response) {
+    response.status(HTTP_RESPONSE_CODE_SERVER_ERROR);
+    return getErrorResponseJson("Error loading predictor! See container logs for details!");
+  }
+
+  private String evaluateRequest(Predictor predictor, Request request)
+      throws PredictorEvaluationException {
+    RequestContentType inputType = RequestContentType.fromValue(request.contentType());
+    switch (inputType) {
+      case Json:
+        {
+          Optional<DataFrame> parsedInput = Optional.<DataFrame>empty();
+          try {
+            parsedInput = Optional.of(DataFrame.fromJson(request.body()));
+          } catch (UnsupportedOperationException e) {
+            throw new PredictorEvaluationException(
+                "This model does not yet support evaluating JSON inputs.");
+          }
+          DataFrame result = predictor.predict(parsedInput.get());
+          return result.toJson();
+        }
+      case Csv:
+        {
+          Optional<DataFrame> parsedInput = Optional.<DataFrame>empty();
+          try {
+            parsedInput = Optional.of(DataFrame.fromCsv(request.body()));
+          } catch (UnsupportedOperationException e) {
+            throw new PredictorEvaluationException(
+                "This model does not yet support evaluating CSV inputs.");
+          }
+          DataFrame result = predictor.predict(parsedInput.get());
+          return result.toCsv();
+        }
+      case Invalid:
+      default:
+        throw new UnsupportedContentTypeException(request.contentType());
```

Comment:
Probably don't need the more specific exception

Reply:
Changed this to `InvalidRequestTypeException`. We now catch this and return a `400` code, in contrast with the previous `500` return code for the `PredictorEvaluationException`.